### ISSUE TITLE
Database Exception while upgrading to 3.6.0

### ISSUE THREAD

**Issue by mikemccarron** [ethercreative/simplemap#173]
### Description

When upgrading to version 3.6.0 the update fails with the follow SQL error: `SQLSTATE[42S22]: Column not found: 1054 Unknown column 'content.field_mapLocation' in 'field list'`

### Steps to reproduce

1. Run updater via CraftCMS backend or run `composer update` after updating version value to 3.6.0

### Additional info

- Craft version: 3.2.2
- Maps version: 3.5.2
- PHP version: 7.3.1
- Database driver & version: MySQL 5.7.25
- Other Plugins: https://cl.ly/670a45e34a1f

**Comment by peteeveleigh** [comment]
I think I have the same problem. I was unable to update with Map installed.
I removed the field which allowed me to update Craft. 
Now trying to install Map 3.6 but it's failing silently. I'm currently looking at the logs to see if there's an indication of why it won't install.

**Comment by PaulCWoods** [comment]
I’m getting the same issue and the same error as the OP. Upgrading to 3.6 broke the site. I’ve reverted to 3.5.2 in the meantime, which, of course, means entries with the map field won’t save.

**Comment by alexjcollins** [comment]
Sorry for the issues everyone, naturally our migration, update and install tests all worked on the sites we tested on 😔.
@fantasticmachine did you find anything useful in your logs by any chance?

**Comment by peteeveleigh** [comment]
@alexjcollins Couldn't really see anything at all in the logs tbh. I'll have another look.

**Comment by peteeveleigh** [comment]
@alexjcollins  Looks like it's throwing an error during the migrations relating to the maps table already existing. 
I've tried deleting the table but get the same error. So my guess is that it's being created then another migration tries to create it again.
Digging further.

Seems that it's running the install migration multiple times, finding the maps table exists and trying to rollback but can't rollback because the savepoint doesn't exist. 🤷‍♂ 

Update!

OK got it installed. Once I'd stopped it trying to recreate the db table I could see that there was an issue porting the existing map data (the original map field had reverted to a plain text type so I guess the migration that moves it to the new version wasn't happy. 
Once I removed my existing map field the installation ran OK.

**Comment by alexjcollins** [comment]
@fantasticmachine Thanks so much for taking the time to look into this; I really appreciate it! @Tam who is lead on the development of this plugin is on annual leave today but will be back in tomorrow, and we'll try and get a fix out for everyone.

**Comment by peteeveleigh** [comment]
@alexjcollins My problems could be related to a really tricky update path. I had a ton of issues updating this site to Craft 3.2 so it's entirely possible that they are the cause of this problem rather than there being something wrong with the plugin itself.

**Comment by mikemccarron** [comment]
I have a pretty standard implementation of the _Maps_ field type, one instance in a Group. I do however have a fairly robust collection of other field types including a complex _Matrix_ field type. There are no _Maps_ in there though.

**Comment by Tam** [comment]
@mikemccarron Did you run the database migrations before getting this error?

**Comment by mikemccarron** [comment]
@Tam I ran the update as described above. I didn't take any additional database migration steps.

**Comment by Tam** [comment]
@mikemccarron When updating via the CP, did you see Craft run any database migrations?

**Comment by mikemccarron** [comment]
@Tam No it doesn't appear to get that far. It errors right after checking server requirements. https://cl.ly/3b066e53a03d

**Comment by Tam** [comment]
@mikemccarron After it's errored can you try manually running `./craft migrate`?

**Comment by mikemccarron** [comment]
@Tam Sure. This was the output:

`Yii Migration Tool (based on Yii v2.0.21)
No new migrations found. Your system is up-to-date.`

**Comment by Tam** [comment]
Is the latest version of Maps definitely installed or did Craft roll it back after the error? If it is installed, try `./craft migrate/all`.

**Comment by KatieMFritz** [comment]
I had a database error that broke the site after upgrading to Maps 3.6.0 too. `./craft migrate/all` fixed it.

**Comment by PaulCWoods** [comment]
`./craft migrate/all` fixed the issue for me too, after requiring 3.6.0

**Comment by mikemccarron** [comment]
@tam I ran `./craft migrate/all` after running `./craft update all` and it said everything was up to date. I downgraded back to 3.5.2. I had a separate issue connecting to my local database `Can't connect to MySQL server on '127.0.0.1' (61) when trying to connect` which had been preventing me from backing up the database, so i changed the`DB_SERVER` in my `.env` to `localhost`. I ran `./craft update all` again and it was successful. I looks like it was a database connection issue all along. Hopefully this helps anyone else having this issue.
